main/views.py

```diff
@@ -2,18 +2,24 @@
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import redirect
 from django.contrib.auth import logout
+from daycare.models import DayCare
+from dogowner.models import DogOwner
+from daycare.views import daycare_home
 
 
 def index(request):
     if request.user.is_authenticated:
         return redirect(to='homepage')
-
     return redirect(to='login')
 
 
 @login_required()
 def homepage(request):
-    return render(request, 'main/homepage.html')
+    if DogOwner.objects.filter(user=request.user).exists():
+        return render(request, 'main/homepage.html')
+    elif DayCare.objects.filter(user=request.user).exists():
+        return daycare_home(request)
+    return about(request)
```

Comment:
do we still need the login_required decorator? if so, who is this last return for

Reply:
Your right, we only have daycare and dog-owner users type so the last return is unnecessary.
fixed